src/QMCWaveFunctions/PlaneWave/PWOrbitalBuilder.cpp

```diff
@@ -113,38 +112,33 @@ std::unique_ptr<WaveFunctionComponent> PWOrbitalBuilder::putSlaterDet(xmlNodePtr
       aAttrib.put(cur);
       if (ref == "0")
         ref = id;
-      int firstIndex = targetPtcl.first(spin_group);
+      const int firstIndex = targetPtcl.first(spin_group);
+      const int lastIndex  = targetPtcl.last(spin_group);
       std::map<std::string, SPOSetPtr>::iterator lit(spomap.find(ref));
-      std::unique_ptr<Det_t> adet;
       //int spin_group=0;
       if (lit == spomap.end())
       {
         app_log() << "  Create a PWOrbitalSet" << std::endl;
         std::unique_ptr<SPOSet> psi(createPW(cur, spin_group));
         spomap[ref] = psi.get();
-        adet        = std::make_unique<Det_t>(std::move(psi), firstIndex);
+        dets.push_back(std::make_unique<DiracDeterminant<>>(std::move(psi), firstIndex, lastIndex));
       }
       else
       {
         app_log() << "  Reuse a PWOrbitalSet" << std::endl;
         std::unique_ptr<SPOSet> psi((*lit).second->makeClone());
-        adet = std::make_unique<Det_t>(std::move(psi), firstIndex);
+        dets.push_back(std::make_unique<DiracDeterminant<>>(std::move(psi), firstIndex, lastIndex));
       }
       app_log() << "    spin=" << spin_group << " id=" << id << " ref=" << ref << std::endl;
-      if (adet)
-      {
-        adet->set(firstIndex, targetPtcl.last(spin_group) - firstIndex);
-        sdet->add(std::move(adet), spin_group);
-      }
       spin_group++;
     }
     cur = cur->next;
   }
 
   if (spin_group)
-    return sdet;
+    return std::make_unique<SlaterDet>(targetPtcl, std::move(dets));;
 
-  APP_ABORT(" Failed to create a SlaterDet at PWOrbitalBuilder::putSlaterDet ");
+  myComm->barrier_and_abort(" Failed to create a SlaterDet at PWOrbitalBuilder::putSlaterDet ");
```

Comment:
See above re errors, myComm etc.

Reply:
This is a builder so just replace APP_ABORT with `myComm->barrier_and_abort`. Overall we need to reduce myComm pollution.